### PR TITLE
add optional and snake_cased options features

### PR DESCRIPTION
**Pull request by matthieudesprez**

### CODE REVIEW

**Comment by matthieudesprez** [comment]
I previously used datamodel-code-generator to generate pydantic model from an OpenApi Schema, and I've found it was convenient to have these Optional and snake_cased handled when generating a lot of pydantic models.

For some months now I'm using a local modified version of your project, I've found it useful and clever in the way you are using the datamodel-code-generator with raw JSON, so I've ended up taking some time for adding these features to your project if it can be useful for someone else.

File: client/src/App.tsx [hunk 1]
```diff
@@ -8,25 +8,29 @@ import "ace-builds/src-noconflict/mode-python";
 import "ace-builds/src-noconflict/theme-monokai";
 
 const apiUrl = "https://ufgjji253b.execute-api.us-east-1.amazonaws.com/prod";
-const defaultJsonObject = '{\n\t"foo": 5 \n}';
+const defaultJsonObject = '{\n\t"foo": 5 \n"barBaz": "hello"}';
```
Comment (brokenloop) [inline client/src/App.tsx:11]:
```suggestion
const defaultJsonObject = '{\n\t"foo": 5, \n\t"barBaz": "hello"}';
```

File: client/src/App.tsx [hunk 2]
```diff
@@ -89,6 +93,33 @@ function App() {
...
+          </div>
+         <div className="field">
+            <p className="option">
+              <label className="checkbox">
+                <input
+                  type="checkbox"
+                  checked={options.snakeCased}
+                  onChange={e => setOptions({...options, snakeCased: e.target.checked})}
+                />
+                Force relevant fields as snake_cased with corresponding aliases
```
Comment (brokenloop) [inline client/src/App.tsx:118]:
```suggestion
                Alias camelCase fields as snake_case
```

**Review by brokenloop — CHANGES_REQUESTED** [review]
Looks great! Some minor suggestions in place, otherwise happy to merge.

File: client/src/App.tsx (hunk 1 above)
Comment (matthieudesprez) [reply client/src/App.tsx:11]: oops, nice catch. Weird that I didn't see it when developing locally

File: client/src/App.tsx (hunk 2 above)
Comment (matthieudesprez) [reply client/src/App.tsx:118]: more concise indeed

**Pull request merged**